code/core/src/core-server/server-channel/open-in-editor-channel.ts

```diff
@@ -0,0 +1,73 @@
+import { join } from 'node:path';
+
+import type { Channel } from 'storybook/internal/channels';
+import { getProjectRoot } from 'storybook/internal/common';
+import type {
+  OpenInEditorRequestPayload,
+  OpenInEditorResponsePayload,
+  RequestData,
+  ResponseData,
+} from 'storybook/internal/core-events';
+import { OPEN_IN_EDITOR_REQUEST, OPEN_IN_EDITOR_RESPONSE } from 'storybook/internal/core-events';
+import { telemetry } from 'storybook/internal/telemetry';
+import type { CoreConfig, Options, StoryIndex } from 'storybook/internal/types';
+
+import launch from 'launch-editor';
+
+export async function initOpenInEditorChannel(
+  channel: Channel,
+  _options: Options,
+  coreOptions: CoreConfig
+) {
+  channel.on(OPEN_IN_EDITOR_REQUEST, async (payload: OpenInEditorRequestPayload) => {
+    const sendTelemetry = (data: { success: boolean; error?: string }) => {
+      if (!coreOptions.disableTelemetry) {
+        telemetry('open-in-editor', data);
+      }
+    };
```

Comment:
I find this abstraction unnecessary

Reply:
does `telemetry()` internally not cancel itself, when `options.disableTelemetry` is truthy?!?